packages/server-boilerplate/src/orchestrator/routes/LogoutRoute.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import { Request } from 'express';
+import { Route } from '../../routes';
+import { EmptyObject } from '../../types';
+
+export class LogoutRoute extends Route<Request<EmptyObject, { success: boolean }>> {
+  async buildResponse() {
+    const { sessionCookie } = this.req;
+    const sessionId = sessionCookie?.id;
+
+    if (sessionId && this.req.sessionModel) {
+      await this.req.sessionModel.deleteById(sessionId);
+    }
+
+    if (sessionCookie !== undefined) {
+      sessionCookie.reset?.();
```

Comment:
You keep pulling nice tricks from your JS/TS hat Rohan 😄 Does this mean that `reset` will be called, only if it is defined **and** a function?

Reply:
Technically I don't think it checks if it's a function, just if `sessionCookie` has a property called `reset`. Should be safe tho 😄

Reply:
You are correct! At a first glance, I initially expected a `sessionCookie?.reset()` syntax for that, but I guess I would be wrong, since

```js
const a = {};
console.log(a.test); // undefined - no error thrown
console.log(a.test()); // throws error: a.test is not a function
```

So the syntax I expected would not help if `reset` itself is not defined, since calling an `undefined` function is invalid!

[Mozilla docs - Optional chaining with function calls](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)